### PR TITLE
Update some naming details in documentation

### PR DESCRIPTION
**Pull request by dustmop**
### Motivation

Questions from traces indicated some inconsistencies in the naming of components in our documentation.

### CODE REVIEW

File: docs/public/components/creating-components.md [hunk 1]
```diff
@@ -121,14 +121,16 @@ The only requirement is that there is a public instantiation function called `Ne
...
     }
     ```
 
-To access an argument in the  `NewComponent` function, use a special struct named `Requires`. This struct returns a special stuct named `Provides`. This internal nomenclature is used to handle the different component dependencies using Fx groups.
+To require input arguments to the `NewComponent` instantiation function, use a special struct named `Requires`.
+The instantiation function returns a special stuct named `Provides`. This internal nomenclature is used
+to handle the different component dependencies using Fx groups.
 
-In this example, the compression component must access the configuration component and the log component. To express this, define a `Requires` struct with two fields. The name is irrelevant, but the type must be a concrete type.
+In this example, the compression component must access the configuration component and the log component. To express this, define a `Requires` struct with two fields. The name of these field is irrelevant, but the type must be the concrete type of the interface you require.
```
Comment (alai97) [inline docs/public/components/creating-components.md:133]:
```suggestion
In this example, the compression component must access the configuration component and the log component. To express this, define a `Requires` struct with two fields. The name of the fields is irrelevant, but the type must be the concrete type of interface that you require.
```

**Review by alai97 — APPROVED** [review]
Looks good, one wording nit!

**Review by GustavoCaso — APPROVED** [review]
LGTM 🎉

**Pull request merged**